Chapter3/README.md

```diff
@@ -0,0 +1,202 @@
+
+- [Node.js](https://nodejs.org/en/download/) must be installed to run this sample.
+- [Dotnet Core SDK](https://dotnet.microsoft.com/download) must be installed to run this sample.
+- *At least* **two** Azure Active Directory (Azure AD) tenants if you want to test deployment end-to-end. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/documentation/articles/active-directory-howto-tenant/).
+- On each tenant, *at least* **one** admin account and **one** non-admin/user account for testing purposes.
+- An **Azure subscription**. This sample uses **Azure App Services** and **Azure Storage**.
+- A modern Browser. This sample uses **ES6** conventions and will not run on **Internet Explorer**.
+- We recommend [VS Code](https://code.visualstudio.com/download) for running and debugging this cross-platform application.
+- We recommend [Azure Tools for VS Code Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-node-azure-pack) for interacting with Azure services directly within VS Code.
+
+## Setup
+
+Using a command line interface such as VS Code integrated terminal, follow the steps below:
+
+### Step 1. Install .NET Core API dependencies
+
+```console
+    cd TodoListAPI
+    dotnet restore
+```
+
+### Step 2. Install Angular SPA dependencies
+
+```console
+    cd ../
+    cd TodoListSPA
+    npm install
+```
+
+## Registration
+
+### Register the service app (TodoListAPI)
+
+Use the same app registration credentials that you've obtained during **chapter 2**.
+
+### Register the client app (TodoListSPA)
+
+Use the same app registration credentials that you've obtained during **chapter 2**.
+
+> If you want to deploy just your SPA, you may also use the registration credentials that you've obtained during **chapter 1**.
+
+## Deployment
+
+### TodoListAPI
+
+We will deploy our web API on [Azure App Services](https://azure.microsoft.com/services/app-service/).
+
+Using a command line interface such as VS Code integrated terminal, follow the steps below:
+
+#### Step 1. Initialize a local Git
+
+```console
+    git init
+```
+
+Once the initialization is done, commit all your files to your local Git. On Visual Studio Code, you can use the **Source Control** panel on the left bar for this.
+
+#### Step 2. Deploy your app
+
+Click on the Azure icon on the left bar in VS Code. Hover your mouse cursor to **App Service** section and you will see an upward-facing arrow icon. Click on it publish your local files to **Azure App Services**
+
+![publish](../Misc/ch3_publish.png)
+
+#### Step 3. Modify your launchSettings.json
+
+1. Open the `TodoListAPI\Properties\launchSettings.json` file.
+2. Find all app keys `applicationUrl` and replace it with the base address of your web API e.g. `my-web-api.azurewebsites.net`.
+3. Find the app key `launchUrl` and replace it with the API endpoint of your web API e.g. `my-web-api.azurewebsites.net/api/todolist`.
+
```

Comment:
Does Step3 and Step2 need to be switched in order? i.e., change the configs before the deployment.

Reply:
good catch!